tests/legacy/unit/commands/test_verbosity.py

```diff
@@ -0,0 +1,64 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2023 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Test verbosity arguments passed to snapcraft_legacy."""
+
+import logging
+from unittest import TestCase, mock
+
+from testtools.matchers import Equals
+
+from . import LifecycleCommandsBaseTestCase
+
+
+class TestVerbosity(LifecycleCommandsBaseTestCase, TestCase):
+    def test_verbosity_debug(self):
+        """Test --enable-developer-debug flag."""
+        with mock.patch("snapcraft_legacy.internal.log.configure") as mock_configure:
+            result = self.run_command(args=["--enable-developer-debug", "pull"])
+
+        self.assertThat(result.exit_code, Equals(0))
+        mock_configure.assert_called_once_with(log_level=logging.DEBUG)
+
+    def test_verbosity_verbose_long(self):
+        """Test --verbose flag."""
+        with mock.patch("snapcraft_legacy.internal.log.configure") as mock_configure:
+            result = self.run_command(args=["--verbose", "pull"])
+
+        self.assertThat(result.exit_code, Equals(0))
+        mock_configure.assert_called_once_with(log_level=logging.INFO)
+
+    def test_verbosity_verbose_short(self):
+        """Test -v flag."""
+        with mock.patch("snapcraft_legacy.internal.log.configure") as mock_configure:
+            result = self.run_command(args=["-v", "pull"])
+
+        self.assertThat(result.exit_code, Equals(0))
+        mock_configure.assert_called_once_with(log_level=logging.INFO)
+
+    def test_verbosity_verbose_long_and_debug_error(self):
+        """Test --verbose and --enabled-developer-debug cannot be used together."""
+        result = self.run_command(
+            args=["--verbose", "--enable-developer-debug", "pull"]
```

Comment:
what about `args=["pull", "--verbose"]`?

Reply:
It works great when using snapcraft at the command line but fails in the unit test:
```
Try 'run pull -h' for help.

Error: No such option: --verbose
```

I think I need to do 2 things:
1. Write the unit tests with pytest rather than using the existing `CommandBaseTest` fixtures.
2. Write spread tests